Makefile

```diff
@@ -466,6 +466,9 @@ $(apidoc_dirs):
 out/doc/api/assets/%: doc/api_assets/% out/doc/api/assets/
 	cp $< $@
 
+out/doc/api/service_worker.js:
+	cp out/doc/api/assets/service_worker.js out/doc/api/service_worker.js
```

Comment:
For ServiceWorker to take control of what's on the page root, it also needs to be placed on the root, meaning not in the `assets` directory.